scalding-parquet-scrooge/src/main/java/com/twitter/scalding/parquet/scrooge/ScroogeReadSupport.java

```diff
@@ -42,4 +58,125 @@ protected MessageType getProjectedSchema(FieldProjectionFilter fieldProjectionFi
+            + "Only one of (%s, %s, %s) should be set.",
+          PARQUET_READ_SCHEMA, STRICT_THRIFT_COLUMN_FILTER_KEY, THRIFT_COLUMN_FILTER_KEY));
+    }
+
+    //set requestedProjections only when it's specified
+    if (partialSchemaString != null) {
+      requestedProjection = getSchemaForRead(fileMessageType, partialSchemaString);
+    } else if (projectionFilter != null) {
+      try {
+        initThriftClassFromMultipleFiles(context.getKeyValueMetadata(), configuration);
+        requestedProjection = getProjectedSchema(projectionFilter);
+      } catch (ClassNotFoundException e) {
+        throw new ThriftProjectionException("can not find thriftClass from configuration", e);
+      }
+    }
+
+    MessageType schemaForRead = getSchemaForRead(fileMessageType, requestedProjection);
+    return new ReadContext(schemaForRead);
+  }
+
+  /**
+   * Method from ReadSupport, copied because it's static
+   *
+   * @param fileMessageType
+   * @param partialReadSchemaString
+   * @return
+   */
+  public static MessageType getSchemaForRead(MessageType fileMessageType, String partialReadSchemaString) {
+    if (partialReadSchemaString == null)
+      return fileMessageType;
+    MessageType requestedMessageType = MessageTypeParser.parseMessageType(partialReadSchemaString);
+    return getSchemaForRead(fileMessageType, requestedMessageType);
+  }
+
+  /**
+   * Updated method from ReadSupport which checks if the projection's compatible instead of a
+   * stricter if the file's schema contains the projection
+   *
+   * @param fileMessageType
+   * @param projectedMessageType
+   * @return
+   */
+  public static MessageType getSchemaForRead(MessageType fileMessageType, MessageType projectedMessageType) {
+    areGroupsCompatible(fileMessageType, projectedMessageType);
+    return projectedMessageType;
+  }
+
+  /**
+   * Method from ThriftReadSupport, copied because it was private
+   */
+  private void initThriftClassFromMultipleFiles(Map<String, Set<String>> fileMetadata, Configuration conf) throws ClassNotFoundException {
+    if (thriftClass != null) {
+      return;
+    }
+    String className = conf.get(THRIFT_READ_CLASS_KEY, null);
+    if (className == null) {
+      Set<String> names = ThriftMetaData.getThriftClassNames(fileMetadata);
+      if (names == null || names.size() != 1) {
+        throw new ParquetDecodingException("Could not read file as the Thrift class is not provided and could not be resolved from the file: " + names);
+      }
+      className = names.iterator().next();
+    }
+    thriftClass = (Class<T>) Class.forName(className);
+  }
+
+  /**
+   * Validates that the requested group type projection is compatible.
+   * This allows the projection schema to have extra optional fields.
+   *
+   * @param fileType
+   * @param projection
+   */
+  public static void areGroupsCompatible(GroupType fileType, GroupType projection) {
```

Comment:
what if we rename this `assertGroupsAreCompatible` to make it clearer it is going to throw rather than return a value.